src/hocs/withTimeoutFallback/index.spec.js

```diff
@@ -0,0 +1,69 @@
+import {mount} from 'enzyme/build/index';
+import React from 'react';
+import sinon from 'sinon';
+import withTimeoutFallback from './index';
+
+const FakeSpinner = () => (<div> ...spinner </div>);
+const FallbackComponent = () => (<div> Alt Fallback </div>);
+
+
+test('Default Spinner with timeout', () => {
+  const clock = sinon.useFakeTimers();
+  const SpinnerWithTimeout = withTimeoutFallback(FakeSpinner, {timeout: 100});
+
+  const component = mount(
+    <SpinnerWithTimeout />,
+  );
+  expect(component).toMatchSnapshot();
+  clock.tick(10001);
+  component.update();
+  expect(component).toMatchSnapshot();
+});
+
+test('Default spinner with shorter timeout', () => {
+  const clock = sinon.useFakeTimers();
+  const SpinnerWithTimeout = withTimeoutFallback(FakeSpinner, {timeout: 100});
+
+  const component = mount(
+    <SpinnerWithTimeout />,
+  );
+
+  expect(component).toMatchSnapshot();
+  clock.tick(100);
+  clock.tick(100);
+  clock.tick(100);
+  clock.tick(100);
```

Comment:
This is probably correct (haven't used this feature before), but just wondering why are you calling `clock.tick` multiple times here?

Reply:
It seems it requires an additional tick after the time expires for it to work properly. I changed it to have one tick the length of the timeout and the second tick the duration of the timer (currently 100ms)